lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js

```diff
@@ -49,12 +49,23 @@ class ExternalAnchorsUseRelNoopenerAudit extends Audit {
       });
     }
 
+    let debugString;
     const pageHost = new URL(artifacts.URL.finalUrl).host;
     // Filter usages to exclude anchors that are same origin
     // TODO: better extendedInfo for anchors with no href attribute:
     // https://github.com/GoogleChrome/lighthouse/issues/1233
+    // https://github.com/GoogleChrome/lighthouse/issues/1345
     const failingAnchors = artifacts.AnchorsWithNoRelNoopener.usages
-      .filter(anchor => anchor.href === '' || new URL(anchor.href).host !== pageHost)
+      .filter(anchor => {
+        try {
+          return new URL(anchor.href).host !== pageHost;
+        } catch (err) {
+          debugString = 'Lighthouse was unable to determine the destination ' +
+              'of some anchor tags. Remove the href attribute if they are not ' +
```

Comment:
ah I didn't realize `anchor.href === ''` would be true even if it didn't have an href at all. This seems like an ambiguous case then. There'd be a false positive most of the time if we warn for all missing `href`s but it's feasible they still send externally.